### PR TITLE
Add application/json content type where required

### PR DESCRIPTION
**Pull request by cgreening**
This seems to be a bit horrible as I've had to add the header everywhere JSON is returned. I don't know if there's a better way to do this.

Fixes: https://github.com/fsouza/fake-gcs-server/issues/451

### CODE REVIEW

File: fakestorage/object.go [hunk 1]
```diff
@@ -421,6 +428,7 @@ func (s *Server) listObjectACL(w http.ResponseWriter, r *http.Request) {
 
 func (s *Server) setObjectACL(w http.ResponseWriter, r *http.Request) {
 	vars := mux.Vars(r)
+	w.Header().Set("Content-Type", "application/json")
```
Comment (fsouza) [inline fakestorage/object.go:431]: can you move it to line 444?

File: fakestorage/object.go [hunk 2]
```diff
@@ -408,6 +413,8 @@ func (s *Server) deleteObject(w http.ResponseWriter, r *http.Request) {
 }
 
 func (s *Server) listObjectACL(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
```
Comment (fsouza) [inline fakestorage/object.go:416]: can you move it to 425?

File: fakestorage/object.go [hunk 3]
```diff
@@ -398,6 +401,8 @@ func (s *Server) getObject(w http.ResponseWriter, r *http.Request) {
 func (s *Server) deleteObject(w http.ResponseWriter, r *http.Request) {
 	vars := mux.Vars(r)
 	err := s.backend.DeleteObject(vars["bucketName"], vars["objectName"])
+	w.Header().Set("Content-Type", "application/json")
```
Comment (fsouza) [inline fakestorage/object.go:404]: move to 412

**Review by fsouza — COMMENTED** [review]
Yeah, this is non-ideal, but if you move the header definitions to right before the response is encoded I can merge it to unblock you (I added inline comments to some examples, but there are other instances).

I can refactor it later to either use a helper function to write the response or wrap all handlers that can return JSON (pretty much all of them except for download? heh)

**Comment by cgreening** [comment]
Let me know if that looks ok - we're not blocked as I can point our docker-compose at my fork for now.

**Comment by fsouza** [comment]
> Let me know if that looks ok - we're not blocked as I can point our docker-compose at my fork for now.

No problem, I'll merge this and tag a release just to get it fixed. Hopefully I can find some time later this week to refactor the code. Thank you very much for contributing!

**Review by fsouza — APPROVED** [review]

**Pull request merged**